pkg/network/node/networkpolicy.go

```diff
@@ -276,15 +285,25 @@ func (np *networkPolicyPlugin) DeleteNetNamespace(netns *osdnv1.NetNamespace) {
 	np.updateMatchCache(npns)
 }
 
-func (np *networkPolicyPlugin) SetUpPod(podIP string) error {
-	otx := np.node.oc.NewTransaction()
-	// when a pod is created isolate the pod until the current network policy rules can be
-	// evaluated
-	otx.AddFlow("table=27, priority=500, cookie=1, ip, nw_src=%s, actions=drop", podIP)
-	otx.AddFlow("table=80, priority=500, cookie=1, ip, nw_src=%s, actions=drop", podIP)
-	if err := otx.Commit(); err != nil {
-		return fmt.Errorf("Error syncing OVS flows to isolate pods %v", err)
+func (np *networkPolicyPlugin) SetUpPod(pod *corev1.Pod, podIP string) error {
+	np.lock.Lock()
+	defer np.lock.Unlock()
+	np.localPodIPs[pod.UID] = podIP
+
+	syncTriggered := np.refreshPodNetworkPolicies(pod)
+	// if network policy update is required for this pod, add isolation rules
+	// syncFlows that will be triggered by refreshPodNetworkPolicies needs to hold np.lock.Lock()
+	// therefore it's safe to add flows after it was triggered
+	if syncTriggered {
+		otx := np.node.oc.NewTransaction()
+		// when a pod is created isolate the pod until the current network policy rules can be evaluated
+		otx.AddFlow("table=27, priority=500, cookie=%d, ip, nw_src=%s, actions=drop", NetPolIsolationCookie, podIP)
+		otx.AddFlow("table=80, priority=500, cookie=%d, ip, nw_src=%s, actions=drop", NetPolIsolationCookie, podIP)
```

Comment:
wait, shouldn't the table 80 rule be `nw_dst=%s` ?
(yeah, it was src in the old code, but that would have been wrong too)

Reply:
huh looks like it